### DISCUSSION TITLE
Start a local music at a random time

### DISCUSSION THREAD

**Discussion by fdecourt** [hklages/node-red-contrib-sonos-plus#303] [Q&A]
Hi,
I love this plugin, thanks for your work.
One of my use of the plugin is to fake presence when I am not at home.
I have a 12 minutes MP3, simulating life at home (people eating, walking, playing music, chatting....)
It is working well, but I would like to randomly start the sounds at different moment of my file (starting at 93 seconds for example and not from the begining of the file). Generating random numbers are easy, but how to inject it to the node ?
Thanks,
regards,
F.

**Comment by hklages** [comment]
Hi.

Nice use case.

You can try group.seek and move to any position in your track.

See
https://github.com/hklages/node-red-contrib-sonos-plus/wiki/A.1-Universal-Node#groupseek

Best regards Henning

**Reply by fdecourt** [reply]
Thanks for the tip, I will try that and keep you informed.
Regards,
F.

**Comment by fdecourt** [comment]
Hi,
It is not working. Here is the message error : 
"group.seek:Sonos error on Seek UPnPError 701 (Transition not available) :: Details: {"stack":"SonosError: Sonos error on Seek UPnPError 701 (Transition not available)\n    at AVTransportService.handleErrorResponse (/config/node_modules/@svrooij/sonos/lib/services/base-service.js:227:23)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at Object.groupSeek [as group.seek] (/config/node_modules/node-red-contrib-sonos-plus/src/sonos-universal.js:1971:5)","message":"Sonos error on Seek UPnPError 701 (Transition not available)","Action":"Seek","FaultCode":"s:Client","Fault":"UPnPError","UpnpErrorCode":701,"UpnpErrorDescription":"Transition not available","name":"SonosError"}"

My local MP3 is hosted on my nas, so I am using group.play.streamhttp.

When I check group.get.actions, it shows only set, stop, play.

Thanks

Regards

**Comment by hklages** [comment]
Right - a stream can not be repositioned.  You have to clear the SONOS Queue, add the mp3 to the SONOS Queue and then you should be able to use group.seek

**Reply by fdecourt** [reply]
Hi,
Thanks for your advices.
I am able to manage and play the queue, but group.seek does not work, without error, just pleying the track from the begining.
With group.get.actions, it shows me that X_DLNA_SeekTime is available (payload: "Set, Stop, Pause, Play, X_DLNA_SeekTime, X_DLNA_SeekTrackNr")
Is it because I am using group.queue.uri, with the URL of my MP3 (http://192.XX.XX.XX:XXXX/local/sounds/simu.mp3)?
![image](https://github.com/user-attachments/assets/d16d7d04-bd97-40cd-8365-5e5804b8fe58)
![image](https://github.com/user-attachments/assets/e73fc0bb-c4de-47ec-aa90-9febe1fe168d)

Thanks for your help.
Regards

**Comment by hklages** [comment]
...please put the group play.group before the seek command.

Does that work?

Grüße Henning Klages

**Reply by fdecourt** [reply]
Hi hHi @hklages ,
Yes it worked perfectly. So I just develop small node to generate a random time in hh:mm;ss (https://flows.nodered.org/node/@fdecourt/node-red-contrib-random-duration). Now, all my usecase is working as I want !
Thanks for your help and your work !

**Comment by hklages** [comment]
Good to hear. Thanks for the feedback.
